### ISSUE TITLE
The AGN radio synchrotron term never reaches the SED: radio_loudness is inert, and it is the root cause of #1461

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1811]
## Summary

The AGN radio synchrotron term does not reach the SED. `radio_loudness` — the parameter that sets its amplitude — is inert at the prediction layer, with its value verified present in the parameter dict and its input `L_4400_intrinsic` verified published and non-zero.

Every layer below the SED checks out. The kernel is correct, the config is correct, the input is published, the parameter arrives. The term still contributes ~3.6e-10 relative.

This is the root cause of #1461: `radio_dpl` cannot differ from `radio_powerlaw` because the term the two models shape is absent from the output.

All measurements on `origin/main` @ `57baf39ed`, 2026-08-15, public API only.

## The layer-by-layer trace

| layer | evidence | verdict |
|---|---|---|
| kernel `radio_total_dpl_terms` | all four shape knobs separate: `alpha1` 5.55e-01, `alpha2` 1.24e+00, `log_nu_t` 4.23e-01, `log_nu_cut` 2.19e-01; kernel `dpl` vs `powerlaw` = 9.38e-01 | ✅ correct |
| config | `spec.radio_agn_model == 'dpl'` for both the legacy `radio={'type':'radio_dpl'}` and composable `radio={'agn':{'type':'dpl'}}` spellings | ✅ correct |
| input | `state.derived['L_4400_intrinsic'] = 4.255850e+29` (composable, multicolor disc + skirtor, `agn_log_lbol=13`) | ✅ published |
| parameter | `params['radio_loudness'] == 4.0` when requested | ✅ arrives |
| **SED** | see below | ❌ **term absent** |

## The measurement

Fixture: SF radio arm disabled (`radio={'sf': {'type': 'none'}}`) so the AGN arm is the only radio contributor; luminous AGN (`log_lbol=13`); one 21 cm band; `predict_photometry`, the surface a fit optimises.

Setting the parameter through the grammar, and then forcing it directly into the parameter dict to rule the grammar out entirely:

```
requested loudness=0.0: params['radio_loudness']=0.0 -> 21cm = 3.0197672652e-14
requested loudness=4.0: params['radio_loudness']=4.0 -> 21cm = 3.0197672663e-14
forced    radio_loudness=0.0 in params -> 21cm = 3.0197672652e-14
forced    radio_loudness=4.0 in params -> 21cm = 3.0197672663e-14
```

Four orders of magnitude in the amplitude parameter move the tenth significant digit.

**The arm is not simply switched off.** Turning the AGN radio block from `'none'` to `'powerlaw'` changes the 21 cm flux by 6.5e10x. But removing the AGN from the model *entirely* — so no `L_4400_intrinsic` exists at all — gives essentially the same number (2.941282e-14 vs 2.941285e-14). So what that block emits is the **free-free** term, which depends on ionizing photons rather than the AGN. The synchrotron term is what is missing.

Consequently every AGN radio shape parameter is inert through the model, while separating cleanly at kernel level:

```
alpha_thin=-1.2  : 0.0000e+00   INERT
alpha_thick=1.5  : 2.8544e-14   INERT
log_nu_t=10.5    : 1.7721e-14   INERT
log_nu_cut=11.5  : 0.0000e+00   INERT
```

## Where it is not

Ruled out by the table above: the kernel, the grammar, the legacy/composable spelling split, `component_factory` (it does pass `agn_radio_model=radio_agn_model`), and `emission_terms`' branch (`model = self.config.agn_radio_model`, then a correct `radio_total_dpl_terms` call with the dpl parameters).

Not traced to a single line, and deliberately not patched blind. The suspect seam is between `emitter_inputs` / `emission_terms` and the SED. `components/radio/component.py:284` reads the input fail-open —

```python
"L_4400_intrinsic": jnp.asarray(derived.get("L_4400_intrinsic", 0.0)),
```

— which is the #1240 shape and would produce exactly this symptom if the radio component consumed `derived` before the AGN published into it. But `predict_state` shows the key present and non-zero in the final state, so ordering needs checking rather than assuming.

The other candidate is the band-response machinery around `EMITTER_PROBE_INPUTS` (`component.py:247`). Its own comment says a term identically zero across both probe draws is zero "for a *structural* reason — `include_freefree=False`, or `radio_loudness = 0` (the default, i.e. radio-quiet)". `radio_loudness` **does** default to 0, so if the band response is built once at build time against that default, the synchrotron term would be baked to zero and no runtime value could revive it — the same "precompute freezes a parameter" shape as #1808. That is a hypothesis, not a measurement.

Whoever fixes it should be aware the seam carries an explicit rank-1 requirement (a term whose *shape*, not just amplitude, responds to a runtime input is not rank-1 and no constant band response exists for it — #1107 cost 13% silently), so a naive fix can regress that property.

## Why this matters

`radio_loudness` is the knob that makes an AGN radio-loud. As shipped, a user can set it anywhere in its range and the radio SED does not move, so any fit that frees it explores a dimension the likelihood cannot see and returns the prior. Radio-loud versus radio-quiet is not a subtle distinction to lose.

Related: #1461 (this is its root cause), #1240 (fail-open `.get` probes for values nothing supplies), #1738 (the epic), #1808 (cached state freezing a free parameter).

**Comment by suchethac** [comment]
**Narrowing: the fail-open input hypothesis is ruled out.** Every input the radio AGN term needs is published and non-zero on the final derived state, measured on the isolated fixture (SF arm off, luminous AGN, z=0.5, physically-broken stellar continuum so the radio band is not contaminated):

```
L_agn_bol          3.828000e+46
L_4400_intrinsic   4.256364e+29
L_2500_intrinsic   5.507072e+29
log_L_agn_bol      4.658297e+01
log_mstar          9.635464e+00
sed_radio          sum = 1.283293e+46   (non-zero — the component IS emitting)
```

So `components/radio/component.py:284`'s `derived.get("L_4400_intrinsic", 0.0)` is not silently taking its default here, and component ordering is not the cause. That was one of the two mechanisms in the report above; it is out.

What remains: `sed_radio` is non-zero with the SF arm disabled, and it is insensitive to `radio_loudness` over four orders of magnitude. Since the free-free term depends on ionizing photons rather than the AGN, the non-zero `sed_radio` is consistent with free-free carrying it while the **synchrotron sub-term** is absent or unscaled. The kernel produces that sub-term correctly when called directly — `radio_total_dpl_terms` returns `{sf, ff, agn}` and all four dpl shape knobs move the total by 0.22–1.24.

Also worth recording, because it rules out the other mechanism I suggested: building with `loudness=Fixed(4.0)` and forcing `params['radio_loudness'] = 4.0` at predict time on a default build give the **same** answer to ten digits (3.0197672663e-14). If the band response were baked at build time against the default `radio_loudness = 0`, those two would differ. They do not, so it is not a build-time baking effect either.

That leaves the seam between `emission_terms`' returned `{sf, ff, agn}` dict and what is summed into `sed_radio`. I have not traced which of the three terms survives, and am not guessing further in the issue.

**Comment by suchethac** [comment]
**Withdrawing this. It is not a defect — the fixture was.** The AGN radio synchrotron term reaches the SED and `radio_loudness` scales it correctly.

I instrumented the component's own `emission_terms` with the model's params and inputs, which is what I should have done before filing:

```
loudness=0.0: sf=0.0000e+00  ff=1.9537e+43  agn=1.0234e+30
loudness=4.0: sf=0.0000e+00  ff=1.9537e+43  agn=1.0234e+34
```

The `agn` term scales by exactly 10⁴ for 4 dex — correct. It was simply **thirteen orders of magnitude below the free-free term**, and therefore invisible in the total.

The reason is in the inputs I printed in the comment above and failed to read critically: **`L_ir = 6.167e+59` erg/s**. A real galaxy is 1e44–1e46. My synthetic SSP had no physical normalization, and free-free tracks the ionizing output, so it inflated with `L_ir` until nothing else could be seen.

Re-measured with the SSP scaled so `L_ir ≈ 6.2e44`:

| | 21 cm flux |
|---|---|
| `radio_loudness = 0` | 1.159065e-27 |
| `radio_loudness = 4` | 1.131507e-23 |
| relative change | **9761×** |

So the conclusion in the report above is void: the term is present, its amplitude knob is live, and the layer-by-layer trace was measuring a term buried under an artifact.

What the trace *did* establish still holds — the kernel is correct, the config is threaded, the inputs are published, and `emission_terms` dispatches properly. All of that was true because none of it was broken.

Closing as invalid. The census fixture in #1807 now carries a physical luminosity scale with this measurement recorded beside the constant, so the same normalization cannot silently produce the same false positive again.

Apologies for the noise. This is the second issue I have filed and withdrawn on this branch, both from fixtures that could not see what I was claiming about.
